### PR TITLE
Feature/cli 24 update nuget pkg in sln

### PR DESCRIPTION
**Pull request by brandongregoryscott**
- Implements feature requested in PR #24 

- Added a `user-prompt` module to allow single access to a readline interface while running a command as well as support utility methods that can be used throughout the project.

- Updated Jest configration and added a test file for this new command. Not sure what else might be needed, or if we want to put the tests in their own directory, but I figured we needed to start somewhere.

Examples of the user-flow:

```
bscot@BSCOTT-PC /c/../Example/ -> and-cli-dev nuget -u
Please enter a package to upgade: AutoMapper
Please enter a version to upgrade 'AutoMapper' to: 7.0.0
[and-cli] Looking for csproj files under the current directory...
[and-cli] Looking for packages matching 'AutoMapper'...
[and-cli] 2 projects found with package 'AutoMapper'.
Continue? (y/n): y
[and-cli] Successfully updated 'AutoMapper' to version 7.0.0. Please check your git status before committing.
```

```
bscot@BSCOTT-PC /c/../Example/ -> and-cli-dev nuget -u
Please enter a package to upgade: DoesNotExist
Please enter a version to upgrade 'DoesNotExist' to: 1.0.0
[and-cli] Looking for csproj files under the current directory...
[and-cli] Looking for packages matching 'DoesNotExist'...
[and-cli] No projects found with package 'DoesNotExist'. Exiting.
```

```
bscot@BSCOTT-PC /c/../Example/ -> and-cli-dev nuget -u
Please enter a package to upgade: AutoMapper
Please enter a version to upgrade 'AutoMapper' to: 7.0.0
[and-cli] Looking for csproj files under the current directory...
[and-cli] Looking for packages matching 'AutoMapper'...
[and-cli] 2 projects found with package 'AutoMapper'.
Continue? (y/n): n
bscot@BSCOTT-PC /c/../Example/ -> 
```

### CODE REVIEW

File: _modules/user-prompt.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Prompts the user for confirmation, and exits if the user does not type y or Y.
+     *
+     * @param {*} question Question to pose to the user
+     * @param {number} [exitStatus=0] Exit status to pass to shelljs if confirmation unsccessful
+     */
+    async confirmOrExit(question, exitStatus = 0) {
+        const prompt = this.getPrompt();
+
+        const answer = await prompt.questionAsync(`${question} ${formatters.yellow("(y/n)")}: `);
```
Comment (wintondeshong) [inline _modules/user-prompt.js]: Super tedious point, but many times its `y/N` to show it accepts lower and uppercase "yY" or "nN"

File: cli-nuget.js [hunk 2]
```diff
@@ -84,6 +88,104 @@ const nugetPublish = {
...
+        const grepResult = shell.grep("-l", this.packageName, csprojFiles);
+        if (grepResult.code !== 0) {
+            echo.error(ERROR_READING_CSPROJ_FILES);
+            shell.exit(grepResult.code);
+        }
+
+        const matchingProjects = grepResult.stdout.split("\n").filter((result) => result.trim() !== "");
+        if (matchingProjects.length === 0) {
+            echo.message(`No projects found with package '${this.packageName}'. Exiting.`);
+            shell.exit(-1);
```
Comment (wintondeshong) [inline cli-nuget.js]: Why negative? I believe '1' is the general catch all conventionally

File: _modules/user-prompt.js (hunk 1 above)
Comment (brandongregoryscott) [reply _modules/user-prompt.js]: Fixed 👍

File: cli-nuget.js (hunk 2 above)
Comment (brandongregoryscott) [reply cli-nuget.js]: Good point 👍 Updated

**Comment by brandongregoryscott** [comment]
@wintondeshong I made those updates, should be good to merge when you have time 👍 Thanks for the feedback!

**Pull request merged**